src/map_field.cpp

```diff
@@ -452,9 +469,11 @@ bool map::process_fields_in_submap( submap *const current_submap,
 
                 // Don't process "newborn" fields. This gives the player time to run if they need to.
                 if( cur.get_field_age() == 0_turns ) {
-                    curtype = field_type_id( "fd_null" );
+                    curtype = fd_null;
                 }
 
+                const field_type &curtype_obj = curtype.obj();
```

Comment:
```suggestion
                const field_type &curtype_obj = *curtype;
```
This is up to personal taste.